pkg/virt-handler/vm.go

```diff
@@ -1677,11 +1677,7 @@ func (c *VirtualMachineController) Run(threadiness int, stopCh chan struct{}) {
 		}
 	}
 
-	heartBeatDone := make(chan struct{})
-	go func() {
-		c.heartBeat.Run(c.heartBeatInterval, stopCh)
-		close(heartBeatDone)
-	}()
+	heartBeatDone := c.heartBeat.Run(c.heartBeatInterval, stopCh)
```

Comment:
I think we need it to guarantee that the `h.labelNodeUnschedulable()` is executed when the `stop` channel is closed.
The VM controller runs the heartBeat through a goroutine and continues its execution.

**-- Without the `heartBeatDone` --**
When the stop channel is closed, both heartBeat goroutine and VM controller continue their execution in parallel:
- the controller tries to end the whole function
- the heartBeat marks the node unschedulable and ends.

This is a race condition since there are two things that could happen:

- the heartBeat is fast enough and ends before the controller ends
   - node is properly marked unschedulable
  
- the controller ends before the heartBeat finishes its job and gets killed since the parent process is dead.
   - the heartBeat may or may have not marked the node unschedulable
   
**-- With the `heartBeatDone` --**
When the stop channel is closed, the VM controller waits for the end of the heartBeat before continue its execution and shutdown.

Reply:
Does it make sense?

Reply:
Yes, it does :) thank you!